EventFilter/EcalRawToDigi/plugins/EcalCPUDigisProducer.cc

```diff
@@ -0,0 +1,196 @@
+  auto const& eedigisProduct = event.get(digisInEEToken_);
+  cms::cuda::ScopedContextAcquire ctx{ebdigisProduct, std::move(taskHolder)};
+  auto const& ebdigis = ctx.get(ebdigisProduct);
+  auto const& eedigis = ctx.get(eedigisProduct);
+
+  // resize tmp buffers
+  dataebtmp.resize(ebdigis.size * EcalDataFrame::MAXSAMPLES);
+  dataeetmp.resize(eedigis.size * EcalDataFrame::MAXSAMPLES);
+  idsebtmp.resize(ebdigis.size);
+  idseetmp.resize(eedigis.size);
+
+  // enqeue transfers
+  cudaCheck(cudaMemcpyAsync(
+      dataebtmp.data(), ebdigis.data.get(), dataebtmp.size() * sizeof(uint16_t), cudaMemcpyDeviceToHost, ctx.stream()));
+  cudaCheck(cudaMemcpyAsync(
+      dataeetmp.data(), eedigis.data.get(), dataeetmp.size() * sizeof(uint16_t), cudaMemcpyDeviceToHost, ctx.stream()));
+  cudaCheck(cudaMemcpyAsync(
+      idsebtmp.data(), ebdigis.ids.get(), idsebtmp.size() * sizeof(uint32_t), cudaMemcpyDeviceToHost, ctx.stream()));
+  cudaCheck(cudaMemcpyAsync(
+      idseetmp.data(), eedigis.ids.get(), idseetmp.size() * sizeof(uint32_t), cudaMemcpyDeviceToHost, ctx.stream()));
+}
+
+void EcalCPUDigisProducer::produce(edm::Event& event, edm::EventSetup const& setup) {
+  // output collections
+  auto digisEB = std::make_unique<EBDigiCollection>();
+  auto digisEE = std::make_unique<EEDigiCollection>();
+  digisEB->resize(idsebtmp.size());
+  digisEE->resize(idseetmp.size());
+
+  // cast constness away
+  // use pointers to buffers instead of move operator= semantics
+  // cause we have different allocators in there...
+  auto* dataEB = const_cast<uint16_t*>(digisEB->data().data());
+  auto* dataEE = const_cast<uint16_t*>(digisEE->data().data());
+  auto* idsEB = const_cast<uint32_t*>(digisEB->ids().data());
+  auto* idsEE = const_cast<uint32_t*>(digisEE->ids().data());
+
+  // copy data
+  std::memcpy(dataEB, dataebtmp.data(), dataebtmp.size() * sizeof(uint16_t));
+  std::memcpy(dataEE, dataeetmp.data(), dataeetmp.size() * sizeof(uint16_t));
+  std::memcpy(idsEB, idsebtmp.data(), idsebtmp.size() * sizeof(uint32_t));
+  std::memcpy(idsEE, idseetmp.data(), idseetmp.size() * sizeof(uint32_t));
+
+  event.put(digisOutEBToken_, std::move(digisEB));
+  event.put(digisOutEEToken_, std::move(digisEE));
+
+  if (produceDummyIntegrityCollections_) {
+    // dummy SRP collections
+    event.emplace(ebSrFlagToken_);
+    event.emplace(eeSrFlagToken_);
+    // dummy integrity for xtal data
+    event.emplace(ebIntegrityGainErrorsToken_);
+    event.emplace(ebIntegrityGainSwitchErrorsToken_);
+    event.emplace(ebIntegrityChIdErrorsToken_);
+    // dummy integrity for xtal data - EE specific (to be rivisited towards EB+EE common collection)
+    event.emplace(eeIntegrityGainErrorsToken_);
+    event.emplace(eeIntegrityGainSwitchErrorsToken_);
+    event.emplace(eeIntegrityChIdErrorsToken_);
+    // dummy integrity errors
+    event.emplace(integrityTTIdErrorsToken_);
+    event.emplace(integrityBlockSizeErrorsToken_);
+  }
```

Comment:
No, I think these are actually consumed by downstream modules (e.g. `PFClusterEMEnergyCorrector`, `EcalDetIdToBeRecoveredProducer`, ...).

So, to avoid producing the dummy collections, we would need to update those consumers and make these collections optional.